elearning-frontend/src/pages/UserSinglePage.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import UserAvatar from '../components/UserAvatar'
+import Button from '../core/Button'
+import Toastify from '../core/Toastify'
+import apiClient from '../services/api'
+
+const UserSinglePage = () => {
+    let { id } = useParams()
+
+    const [user, setUser] = useState(null)
+    const [loading, setLoading] = useState(true)
+    
+    const getUser = (async () => {
+        const data = await apiClient({
+            method: 'get',
+            url: `/api/user/${id}`
+        }).then(response => {
+            setUser({...response.data})
+            setLoading(false)
+        }).catch(error => {
+            Toastify('error', error)
+        })
+    })
+
+    useEffect(() => {
+        getUser()
+    }, [])
```

Comment:
```suggestion
    useEffect(() => {
        apiClient({
            method: 'get',
            url: `/api/user/${id}`
        }).then(response => {
            setUser({...response.data})
            setLoading(false)
        }).catch(error => {
            Toastify('error', error)
        })
    }, [])
```

You have 3 options.

First, you can do this suggested above because creating new function is extra step.

2. But if you want to separate to get more of a context of what is trying to do with the API. You can remove async await part because you are doing the logic part with a promise style. Not async await. Also you dont need to spread when setting the user state because there is only 1 object unless you will change or add new data.


```suggestion
     const getUser = (async () => {
        apiClient({
            method: 'get',
            url: `/api/user/${id}`
        }).then(response => {
            setUser(response.data)
            setLoading(false)
        }).catch(error => {
            Toastify('error', error)
        })
    })
    
    useEffect(() => {
        getUser()
    }, [])
```

3. If you want to use async/await. it should be like this
```suggestion
      const getUser = () => {
        const data = apiClient({
            method: 'get',
            url: `/api/user/${id}`
        })
        
        if (response.status === '200') {
        setUser(response.data)
        }else{
        toastify....
        }
    }
    
    useEffect(() => {
        getUser()
    }, [])
```